packages/contracts/contracts/engine/V2/GenArt721Minter_PBAB.sol

```diff
@@ -251,6 +273,14 @@ contract GenArt721Minter_PBAB is ReentrancyGuard {
                 )
             ) != keccak256(abi.encodePacked("ETH"))
         ) {
+            uint256 pricePerTokenInWei = genArtCoreContract
+                .projectIdToPricePerTokenInWei(_projectId);
```

Comment:
for gas efficiency, can replace subsequent calls to `genArtCoreContract.projectIdToPricePerTokenInWei` with this cached value